File: spring-email-java/src/main/java/com/github/migangqui/spring/email/config/EmailSenderConfig.java
package com.github.migangqui.spring.email.config;

import com.github.migangqui.spring.email.service.EmailSender;
import com.github.migangqui.spring.email.service.EmailSenderImpl;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.mail.javamail.JavaMailSender;

@Configuration
public class EmailSenderConfig {

    @Bean
    public EmailSender emailSender(final JavaMailSender javaMailSender) {
        return new EmailSenderImpl(javaMailSender);
    }
}

File: spring-email-kotlin/src/main/kotlin/com/github/migangqui/spring/email/config/EmailSenderConfig.kt
package com.github.migangqui.spring.email.config

import com.github.migangqui.spring.email.service.EmailSender
import com.github.migangqui.spring.email.service.EmailSenderImpl
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.mail.javamail.JavaMailSender

@Configuration
class EmailSenderConfig {
    
    @Bean
    fun emailSender(javaMailSender: JavaMailSender): EmailSender {
        return EmailSenderImpl(javaMailSender)
    }
}